payments/src/main/java/it/gov/pagopa/payments/service/PaymentsService.java

```diff
@@ -130,15 +139,57 @@ public PaymentsResult<ReceiptEntity> getOrganizationReceipts(@Positive Integer l
 			return this.getGPDCheckedReceiptsList(table, result);
 		}
 		catch (InvalidKeyException | URISyntaxException | StorageException e) {
-			log.error("[getReceiptByOrganizationFCAndIUV] Payments Generic Error " + String.format(LOG_BASE_PARAMS_DETAIL, organizationFiscalCode, debtor), e);
+			log.error("[getReceiptByOrganizationFCAndIUV] Payments Generic Error " + String.format(LOG_BASE_PARAMS_DETAIL, organizationFiscalCode, debtor, service), e);
 			throw new AppException(AppError.RETRIEVAL_RECEIPTS_FAILED, organizationFiscalCode, debtor);
 		}
 
     }
-
+	
+	public void checkGPDDebtPosStatus (CloudTable table, ReceiptEntity receipt) {
+		// the check on GPD is necessary if the status of the receipt is different from PAID
+		if(!receipt.getStatus().trim().equalsIgnoreCase(Status.PAID.name())) {
+			PaymentsModelResponse paymentOption = gpdClient.getPaymentOption(receipt.getPartitionKey(), receipt.getRowKey());
+			if (null != paymentOption && !PaymentOptionStatus.PO_PAID.equals(paymentOption.getStatus())) {
+				throw new AppException(AppError.UNPROCESSABLE_RECEIPT, paymentOption.getStatus(), receipt.getPartitionKey(), receipt.getRowKey());
+			}
+			// if no exception is raised the status on GPD is correctly in PAID -> for congruence update receipt status
+			receipt.setStatus(Status.PAID.name());
+			TableOperation updateOperation = TableOperation.merge(receipt);
+	        try {
+				table.execute(updateOperation);
+			} catch (StorageException e) {
+				log.error("[checkGPDDebtPosStatus] Non-blocking error: "
+						+ "Exception during the update status in table "+ receiptsTable +" for ReceiptEntity [pk:"+receipt.getPartitionKey()+", rk:"+receipt.getRowKey()+"]", e);
+			}   
+		}	
+	}
+	
+	public PaymentsResult<ReceiptEntity> getGPDCheckedReceiptsList(CloudTable table, PaymentsResult<ReceiptEntity> result){
+		// for all the receipts in the azure table, only those that have been already PAID status or are in PAID status on GPD are returned
+		List<ReceiptEntity> checkedReceipts = new ArrayList<>();
+		for (ReceiptEntity re: result.getResults()) {
+			try {
+				this.checkGPDDebtPosStatus(table, re);
+				checkedReceipts.add(re);
+			} 
+			catch (FeignException.NotFound e ) {
+				log.error("[getGPDCheckedReceiptsList] Non-blocking error: "
```

Comment:
ok you'd try to open pr on https://github.com/pagopa/pagopa-infra to create something alarm like [that](https://github.com/pagopa/pagopa-infra/blob/d9106f6a19620f4af7f8cd7ac908f0247e18bde5/src/canoneunico.tf#L273)